app/javascript/dashboard/routes/dashboard/commands/commandbar.vue

```diff
@@ -72,6 +89,49 @@ export default {
       });
       this.setCommandbarData();
     },
+    setModalObserver() {
+      this.disposeModalObserver();
+
+      const ninjaKeysElement = document.querySelector('ninja-keys');
+      if (ninjaKeysElement && ninjaKeysElement.shadowRoot) {
+        const modalElement =
+          ninjaKeysElement.shadowRoot.querySelector('.modal');
+        if (modalElement instanceof HTMLElement) {
+          this.createModalObserver(modalElement);
+        }
+      }
+    },
+    createModalObserver(modalElement) {
+      // Initialize an observer to detect when the command bar modal is closed and opened
+      // The ninja-keys component does not emit a close event https://github.com/ssleptsov/ninja-keys?tab=readme-ov-file#events.
+      // So by MutationObserver we can detect the modal visibility state by observing DOM changes.
+      // By observing changes to the 'class' attribute of the modal element,
+      // we can determine the modal's visibility state.
+
+      this.modalObserver = new MutationObserver(mutations => {
```

Comment:
@pranavrajs It seems like the `ninja-keys` package does not provide a dedicated event for modal close actions. 
I think the `MutationObserver` approach is a reasonable workaround.